core/data/src/main/java/uk/gov/gchq/gaffer/data/elementdefinition/view/View.java

```diff
@@ -160,6 +162,22 @@ public void expandGlobalDefinitions() {
         }
     }
 
+    @JsonIgnore
+    public Function<Element, Set<Object>> getGroupByFunction() {
+        return element -> {
+            Set<Object> key = new HashSet<>();
+            String group = element.getGroup();
+            Set<String> groupBy = getElementGroupBy(group);
+            if (groupBy == null) {
+                return key;
```

Comment:
You could optimise this by not creating the HashSet unless you actually need it.

In the case the groupBy is null you could return Collections.emptySet();

When you create the HashSet you can create it using the size of the groupBy set.